src/Jackalope/Query/Row.php

```diff
@@ -162,7 +162,7 @@ public function getValue($columnName)
         // single-valued properties. We join the values when it's a string
         // for multi-values boolean/binary values we can't provide a
         // defined result so we return null
-        if (is_array($value)) {
+        if (is_array($value) && count($value)) {
             if (is_scalar($value[0]) && !is_bool($value[0])) {
```

Comment:
sorry. i meant to move count() into this if, so it becomes `if count($value) && is_scalar($value[0]) && !is_bool($value[0])) {`.

we should return null for an empty array.